.github/workflows/publish-to-branch.yaml

```diff
@@ -0,0 +1,28 @@
+name: Publish to branch
+
+on:
+  pull_request:
+    branches:
+      - master
+      - main
+      - track/**
```

Comment:
I think this can be removed, but you should confirm removing it will result in the behaviour you expect.  I think in the other kubeflow repos, we only have `branches` restrictions on `push` events, but not on `pull_request` events.  But maybe we hadn't considered it in the setting of the branch publishing, so I might be missing something

Reply:
I see that in many of our repos on `pull_request` events, so maybe we should keep it? For example, if I made a pr to your branch, I suppose it doesn't mean that the publish action should run

Reply:
I agree with @natalian98. PRs to feature branches should not lead to publishing.